tests/client.test.ts

```diff
@@ -0,0 +1,334 @@
+import { workflow_status } from '../schemas/system_db_schema';
+import { DBOS, DBOSConfig, DBOSClient, WorkflowQueue } from '../src';
+import { globalParams, sleepms } from '../src/utils';
+import { generateDBOSTestConfig, setUpDBOSTestDb } from './helpers';
+import { Client } from 'pg';
+
+const _queue = new WorkflowQueue('testQueue');
+
+class ClientTest {
+  @DBOS.workflow()
+  static async enqueueTest(
+    numVal: number,
+    strVal: string,
+    objVal: { first: string; last: string; age: number },
+  ): Promise<string> {
+    return Promise.resolve(`${numVal}-${strVal}-${JSON.stringify(objVal)}`);
+  }
+
+  @DBOS.workflow()
+  static async sendTest(topic?: string) {
+    return await DBOS.recv<string>(topic, 60);
+  }
+
+  @DBOS.workflow()
+  static async eventTest(key: string, value: string, update: boolean = false) {
+    await DBOS.setEvent(key, value);
+    await DBOS.sleepSeconds(5);
+    if (update) {
+      await DBOS.setEvent(key, `updated-${value}`);
+    }
+    return `${key}-${value}`;
+  }
+}
+
+describe('DBOSClient', () => {
+  let config: DBOSConfig;
+
+  beforeAll(async () => {
+    config = generateDBOSTestConfig();
+    await setUpDBOSTestDb(config);
+  });
+
+  beforeEach(async () => {
+    DBOS.setConfig(config);
+    await DBOS.launch();
+  });
+
+  afterEach(async () => {
+    await DBOS.shutdown();
+  });
+
+  test('DBOSClient-enqueue-appVer-notSet)', async () => {
+    const client = new DBOSClient(config.poolConfig, config.system_database);
+    const wfid = `client-enqueue-${Date.now()}`;
+
+    try {
+      type InvokeTest = typeof ClientTest.enqueueTest;
+
+      await client.init();
+      await client.enqueue<Parameters<InvokeTest>>(
+        {
+          workflowName: 'enqueueTest',
+          workflowClassName: 'ClientTest',
+          queueName: 'testQueue',
+          workflowUUID: wfid,
+        },
+        42,
+        'test',
+        { first: 'John', last: 'Doe', age: 30 },
+      );
+
+      const handle = DBOS.retrieveWorkflow<Awaited<ReturnType<InvokeTest>>>(wfid);
```

Comment:
Are we always going to have to type `retrieveWorkflow` like this?

Reply:
No, you can specify the type directly. I could add the `Awaited` utility type to the `retrieveWorkflow ` signature since it is a no op for non promise types.

Reply:
Note, this is for `DBOS.retrieveWorkflow` which is not part of the client. 

It doesn't appear to matter much. w/o the `Awaited` call, the handle type is `WorkflowHandle<Promise<String>>` which makes the return type of `handle.getResult` `Promise<Promise<string>>`. But `await`ing a `Promise<Promise<string>>` results in `string` so it's not that huge a deal. However, adding `Awaited` to both client and `DBOS` versions of `retrieveWorkflow` seems like the right thing to do here.